api4/channel.go

```diff
@@ -1150,9 +1150,17 @@ func getChannelByNameForTeamName(c *Context, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	if !c.App.SessionHasPermissionToChannel(*c.App.Session(), channel.Id, model.PERMISSION_READ_CHANNEL) {
-		c.Err = model.NewAppError("getChannelByNameForTeamName", "app.channel.get_by_name.missing.app_error", nil, "teamId="+channel.TeamId+", "+"name="+channel.Name+"", http.StatusNotFound)
-		return
+	if channel.Type == model.CHANNEL_OPEN {
+		if !c.App.SessionHasPermissionToTeam(*c.App.Session(), channel.TeamId, model.PERMISSION_READ_PUBLIC_CHANNEL) &&
+			!c.App.SessionHasPermissionToChannel(*c.App.Session(), channel.Id, model.PERMISSION_READ_CHANNEL) {
+			c.SetPermissionError(model.PERMISSION_READ_PUBLIC_CHANNEL)
```

Comment:
Any reason the handling of errors in the 2 cases are different? Calling `SetPermissionError` does not capture the teamId and channelName.

Reply:
> Any reason the handling of errors in the 2 cases are different? Calling SetPermissionError does not capture the teamId and channelName.

Not really, my intent was to just to update `getChannelByNameForTeamName` to behave the same (re permissions) as `getChannelByName`. I'm not sure why permissions were handled differently. 

Same for the comment above. I'd prefer to keep the code, behaviour and logging the same, but I'm open to updating/cleaning up this method.

Reply:
Understood. The refactoring was just a suggestion anyways.